### PR TITLE
add oidc samples c#

### PR DESCRIPTION
**Pull request by KanchanaJayasinghe**
Adding C# sample for pickup manager scenario and the related SDK

### CODE REVIEW

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        private const string code = "code";
+        private const string state = "state";
+        private const string tokenRequestBody = "code={0}&redirect_uri={1}&client_id={2}&code_verifier={3}" +
+                "&client_secret={4}&scope=&grant_type=authorization_code";
+        private const string methodPost = "POST";
+        private const string methodGet = "GET";
+        private const string contentType = "application/x-www-form-urlencoded";
+        private const string accept = "Accept=text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8";
+        private const string postRedirectURI = "{0}?id_token_hint={1}&post_logout_redirect_uri={2}";
+
```
Comment (JKAUSHALYA) [inline oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs]: Remove unwanted spaces.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        public static string MethodGet => methodGet;
+
+        public static string ContentType => contentType;
+
+        public static string Accept => accept;
+
+        public static string CodeChallengeMethod => codeChallengeMethod;
+    }
+
```
Comment (JKAUSHALYA) [inline oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs]: Remove unwanted spaces.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/ServerConfigurationManager.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            }
+        }
+
+        // Enable ssl veification to establish trust relationship for the SSL/TLS secure channel.
+        public void EnableSSLverification()
+        {
+            ServicePointManager.ServerCertificateValidationCallback = delegate (object s, X509Certificate
+                certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors) { return true; };
+        }
+
```
Comment (JKAUSHALYA) [inline oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/ServerConfigurationManager.cs]: Remove unwanted spaces.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
+C:\Users\wso2\source\repos\Project - IAM\org.wso2.identity.sdk.oidc\org.wso2.identity.sdk.oidc\bin\Release\netstandard2.0\org.wso2.identity.sdk.oidc.deps.json
```
Comment (JKAUSHALYA) [inline oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt]: We cannot have absolute paths. Shall we try to move them to relative paths?

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.cache [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "version": 1,
+  "dgSpecHash": "Q4JtxSZaPcgDeEOnBoIo+COG/pbcUuBOCxR2v1VtDPkUSIJMQH0FVVFAEn44dSq0hDpgmo071lgsT8S6M5zuZw==",
+  "success": true
+}
```
Comment (JKAUSHALYA) [inline oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.cache]: Add a new line at the end.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.dgspec.json [hunk 6]
```diff
@@ -0,0 +1,66 @@
+{
+  "format": 1,
+  "restore": {
+    "C:\\Users\\wso2\\source\\repos\\Project - IAM\\org.wso2.identity.sdk.oidc\\org.wso2.identity.sdk.oidc\\org.wso2.identity.sdk.oidc.csproj": {}
```
Comment (JKAUSHALYA) [inline oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.dgspec.json]: We can't have absolute paths. Shall we check a way to have relative paths.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs (hunk 1 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs]: Updated the PR.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/ServerConfigurationManager.cs (hunk 3 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/ServerConfigurationManager.cs]: Updated the PR

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs (hunk 2 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/Constants.cs]: Updated the PR.

File: oidc-samples/OIDC-SDK/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt [hunk 7]
```diff
@@ -0,0 +1,76 @@
+C:\Users\wso2\source\repos\Project - IAM\org.wso2.identity.sdk.oidc\org.wso2.identity.sdk.oidc\bin\Release\netstandard2.0\org.wso2.identity.sdk.oidc.deps.json
```
Comment (JKAUSHALYA) [inline oidc-samples/OIDC-SDK/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt]: This file still contains absolute paths. According to [1], we don't need to commit these files.

[1] https://stackoverflow.com/questions/559045/what-are-all-these-filelistabsolute-txt-files-for

**Comment by JKAUSHALYA** [comment]
We don't need to commit binary files like .cache, .pdb, etc... Remove those [1].
Furthermore, we don't need to commit release and debug folders as well. Do a project clean and a NuGet [2] clean before committing. 

[1] https://stackoverflow.com/questions/4618950/what-is-resolveassemblyreference-cache
[2] https://docs.microsoft.com/en-us/nuget/consume-packages/managing-the-global-packages-and-cache-folders

**Comment by JKAUSHALYA** [comment]
Please use the imperative style to commit messages. Please follow the [1] for commit message guidelines. And don't use separate commits for code cleanup. Do an amend. 

[1] https://chris.beams.io/posts/git-commit/

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt (hunk 4 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt]: Remove those files.

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.cache (hunk 5 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.cache]: Removed(This file generated while the build process).

File: oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.dgspec.json (hunk 6 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/org.wso2.identity.sdk.oidc/org.wso2.identity.sdk.oidc/obj/org.wso2.identity.sdk.oidc.csproj.nuget.dgspec.json]: Fixed.

File: oidc-samples/OIDC-SDK/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt (hunk 7 above)
Comment (KanchanaJayasinghe) [reply oidc-samples/OIDC-SDK/org.wso2.identity.sdk.oidc/obj/Release/netstandard2.0/org.wso2.identity.sdk.oidc.csproj.FileListAbsolute.txt]: Resolved.

**Comment by KanchanaJayasinghe** [comment]
Completed requested changes.

**Pull request merged**